### DISCUSSION TITLE
2302 output is double the size of 2301

### DISCUSSION THREAD

**Discussion by rp1231** [LSPosed/MagiskOnWSALocal#423] [Q&A] [answered]
The 2301 output was around 2GB but the 2302 output is around 4GB,
is that because 2302 is larger or because of the new method used for 2302?

**Comment by yujincheng08** [comment, marked as answer]
Aha, this is because 2302 uses VHDX now. And as our script is running in Linux, we have only limited ways to manipulate VHDX, and thus the generated VHDX is large. In windows, you can try `Optimize-VHD system.vhdx`, and the `system.vhdx` can reduce half of its size. But in Linux, we don't have tools to "optimize" it.

**Comment by rp1231** [comment]
Could you explain more in detail how to do that (If it's not that complicated) ?
Could there maybe be a mini script to enable people on windows to do that?

**Reply by yujincheng08** [reply]
`Optimize-VHD` requires Hyper-V installed, and we don't expect all users have this installed, so we won't have such a script.

**Reply by rp1231** [reply]
Thanks, It reduced the file size of system.vhdx from 2GB to 1GB.
Can I use this command on all the vhdx files?

**Reply by Howard20181** [reply]
Yes

**Reply by yujincheng08** [reply]
yup

**Reply by Howard20181** [reply]
e.g.  `Optimize-VHD  "D:\WSA\*.vhdx" -Mode Full`

**Reply by rp1231** [reply]
Thanks a lot!
